File: app/src/main/java/com/saizad/mvvmexample/MVVMExampleCurrentUser.java
package com.saizad.mvvmexample;

import android.content.SharedPreferences;

import androidx.annotation.Nullable;

import com.google.gson.Gson;
import com.saizad.mvvm.CurrentUserType;
import com.saizad.mvvm.model.UserInfo;

public class MVVMExampleCurrentUser extends CurrentUserType<UserInfo> {

    public MVVMExampleCurrentUser(SharedPreferences sharedPreferences, Gson gson) {
        super(new UserInfo(), sharedPreferences, gson);
    }

    @Override
    protected Class<UserInfo> getClassType() {
        return UserInfo.class;
    }

    @Nullable
    @Override
    public String getToken() {
        return null;
    }
}

File: app/src/main/java/com/saizad/mvvmexample/MVVMExampleCurrentUser.kt
package com.saizad.mvvmexample

import com.saizad.mvvm.CurrentUserType
import com.saizad.mvvm.DataStoreWrapper
import com.saizad.mvvmexample.models.ReqResUser
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class MVVMExampleCurrentUser @Inject constructor(dataStoreWrapper: DataStoreWrapper) :
    CurrentUserType<ReqResUser>(dataStoreWrapper) {

    override val classType: Class<ReqResUser>
        get() = ReqResUser::class.java

    override val token: String?
        get() = null
}